tests/core/test_utils.py

```diff
@@ -249,3 +251,86 @@ def test_unexpected_path_dashboard_url(self, dashboard_url):
             click.exceptions.ClickException, match="got an unexpected path"
         ):
             api_to_dashboard_url(dashboard_url)
+
+
+@patch("ggshield.core.utils.load_dotenv")
+@pytest.mark.parametrize(
+    ["cwd", "env", "expected_dotenv_path"],
+    [
+        ("sub", {}, None),
+        (".", {}, ".env"),
+        (".", {"GITGUARDIAN_DONT_LOAD_ENV": "1"}, None),
+        (".", {"GITGUARDIAN_DOTENV_PATH": ".custom-env"}, ".custom-env"),
+    ],
+)
+def test_load_dot_env(
+    load_dotenv_mock: Mock,
+    monkeypatch,
+    tmp_path,
+    cwd: str,
+    env: Dict[str, str],
+    expected_dotenv_path: Optional[str],
+):
+    """
+    GIVEN a file hierarchy like this:
+    /sub/
+    /.env
+    /.custom-env
+    AND environment variables set according to `env`
+    AND the current working directory being `/sub/`
+    WHEN load_dot_env() is called
+    THEN the appropriate environment file is loaded
+    """
+    (tmp_path / "sub").mkdir()
+    (tmp_path / ".env").touch()
+    (tmp_path / ".custom-env").touch()
+
+    if env:
+        for key, value in env.items():
+            monkeypatch.setenv(key, value)
+
+    with cd(str(tmp_path / cwd)):
+        load_dot_env()
+
+        if expected_dotenv_path:
+            expected_dotenv_path = tmp_path / expected_dotenv_path
+            load_dotenv_mock.assert_called_once_with(ANY, override=True)
+            path = load_dotenv_mock.call_args.args[0]
+            path = os.path.abspath(path)
+            assert path == str(expected_dotenv_path)
+        else:
+            load_dotenv_mock.assert_not_called()
+
+
+@patch("ggshield.core.utils.get_git_root")
+@patch("ggshield.core.utils.is_git_dir")
+@patch("ggshield.core.utils.load_dotenv")
+def test_load_dot_env_loads_git_root_env(
+    load_dotenv_mock: Mock, is_git_dir_mock, get_git_root_mock, tmp_path
+):
+    """
+    GIVEN a git repository checkout with this file hierarchy:
+    /sub1/sub2
+    /sub1/.env
+    /.env
+    AND the current working directory being `/sub1/sub2`
+    WHEN load_dot_env() is called
+    THEN the .env file at the root of the git repository is loaded
+    """
+    is_git_dir_mock.return_value = True
+    get_git_root_mock.return_value = str(tmp_path)
+
+    sub_dir = tmp_path / "sub1" / "sub2"
+    git_root_dotenv = tmp_path / ".env"
+
+    sub_dir.mkdir(parents=True)
+    (tmp_path / "sub1" / "sub2").touch()
```

Comment:
Shouldn't this be `(tmp_path / "sub1" / ".env").touch()` to reflect the hierarchy you describe in the docstring ?

Reply:
Oups, good catch!